java/dataset/src/main/java/org/apache/arrow/dataset/scanner/FragmentScanOptions.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.dataset.scanner;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+import org.apache.arrow.dataset.substrait.StringMapNode;
+
+import com.google.protobuf.Any;
+
+import io.substrait.proto.AdvancedExtension;
+
+public interface FragmentScanOptions {
+  String typeName();
+
+  int fileFormatId();
+
+  ByteBuffer serialize();
+
+  /**
+   * serialize the map.
+   *
+   * @param config config map
+   * @return bufer to jni call argument, should be DirectByteBuffer
+   */
+  default ByteBuffer serializeMap(Map<String, String> config) {
```

Comment:
Is this function just written to pass a Java Map to C++ via JNI?

Reply:
Yes

Reply:
Can we put it as a private static helper somewhere? No need to expose it publicly as an instance method